tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -3056,10 +3069,19 @@ bool CudnnSupport::GetConvolveAlgorithms(
     algo_types.push_back(CUDNN_CONVOLUTION_FWD_ALGO_WINOGRAD_NONFUSED);
   }
 
+  if (Deterministic()) {
```

Comment:
It would need to be:

```
if (Deterministic()) {
  use_tensor_ops = cc_major >= 7 && CUDNN_VERSION >= 7000 && TensorOpMathEnabled();
  out_algorithms>push_back({CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_PRECOMP_GEMM,
                                                   use_tensor_ops});
  return true;
}
```
I was trying to not repeat the expression `cc_major >= 7 && CUDNN_VERSION >= 7000 && TensorOpMathEnabled()` but I now appreciate that this is already repeated three times in the file. I'm considering factoring this.

Meanwhile, do multiple commits on the same branch/pull request get squashed on the merge. I'll assume, unless you tell me otherwise, that it doesn't matter if I keep adding new commits (rather than force-pushing).

Reply:
> I was trying to not repeat the expression cc_major >= 7 && CUDNN_VERSION >= 7000 && TensorOpMathEnabled() but I now appreciate that this is already repeated three times in the file. I'm considering factoring this.

Good observation. Maybe factor all of them out as `AllowTensorOpMath()`?

> Meanwhile, can you tell me if multiple commits on the same branch/pull-request get squashed on the merge? I'll assume, unless you tell me otherwise, that it doesn't matter if I keep adding new commits (rather than force-pushing).

All commits in the same PR get squashed.

Reply:
Ah I just saw the second commits. `TensorOpMathAvailable()` is a better name.